### PR TITLE
coprocessor/dag/executor: handle only one row in stream aggregation

### PR DESCRIPTION
**Pull request by zimulala**
#2714 doesn't pass mysql_test.

### CODE REVIEW

**Comment by zimulala** [comment]
/run-integration-common-test tidb-test=pr/453 tidb=pr/5725 tidb-private-test=pr/453

**Comment by zimulala** [comment]
PTAL  @hicqu @winoros

**Comment by zimulala** [comment]
PTAL @AndreMouche @hicqu

File: src/coprocessor/dag/executor/aggregation.rs [hunk 1]
```diff
@@ -241,7 +242,7 @@ impl Executor for StreamAggExecutor {
         // If there is no data in the t, then whether there is 'group by' that can affect the result.
         // e.g. select count(*) from t. Result is 0.
         // e.g. select count(*) from t group by c. Result is empty.
-        if self.count == 0 && !self.group_by_exprs.is_empty() {
+        if !self.has_data && !self.group_by_exprs.is_empty() {
```
Comment (AndreMouche) [inline src/coprocessor/dag/executor/aggregation.rs]: Any test cases to cover `self.has_data=false`?

Comment (zimulala) [reply src/coprocessor/dag/executor/aggregation.rs]: These test cases in TiDB.
Now I add it here.

**Comment by zimulala** [comment]
/run-integration-common-test tidb-test=pr/453 tidb=pr/5725 tidb-private-test=pr/453

**Comment by zimulala** [comment]
/run-integration-common-test tidb-test=pr/453 tidb=pr/5725 tidb-private-test=pr/453

**Comment by zimulala** [comment]
/run-integration-common-test tidb-test=pr/453 tidb=pr/5725 tidb-private-test=pr/453

**Comment by zimulala** [comment]
/run-integration-common-test tidb-test=pr/453 tidb=pr/5725 tidb-private-test=pr/453

**Comment by zimulala** [comment]
/run-integration-common-test tidb-test=pr/453 tidb=pr/5725 tidb-private-test=pr/453

**Comment by zimulala** [comment]
PTAL @AndreMouche

File: src/coprocessor/dag/executor/aggregation.rs [hunk 2]
```diff
@@ -508,6 +498,98 @@ mod test {
...
+        while let Some(row) = agg_ect.next().unwrap() {
+            row_data.push(row.data);
+        }
+        assert_eq!(row_data.len(), expect_row_cnt);
+        let expected_counts = vec![idx_row_cnt, expect_row_cnt as i64];
+        let mut counts = Vec::with_capacity(2);
+        agg_ect.collect_output_counts(&mut counts);
+        assert_eq!(expected_counts, counts);
+
+        // test no row
```
Comment (AndreMouche) [inline src/coprocessor/dag/executor/aggregation.rs]: Is this comment expired?

**Pull request merged**